store/rootmulti/store.go

```diff
@@ -401,23 +401,36 @@ func (rs *Store) TracingEnabled() bool {
 	return rs.traceWriter != nil
 }
 
-// AddListeners adds listeners for a specific KVStore
-func (rs *Store) AddListeners(key types.StoreKey, listeners []types.WriteListener) {
-	if ls, ok := rs.listeners[key]; ok {
-		rs.listeners[key] = append(ls, listeners...)
-	} else {
-		rs.listeners[key] = listeners
+// AddListeners adds state change listener for a specific KVStore
+func (rs *Store) AddListeners(keys []types.StoreKey) {
+	listener := types.NewMemoryListener()
+	for i := range keys {
+		rs.listeners[keys[i]] = listener
```

Comment:
This doesn't add a listener for provided keys, it overwrites whatever listener exists for those keys with a brand new and empty listener, which is a destructive (and breaking) change.

Reply:
fixed to add listeners only if one doesn't exist for that key.